base/src/main/java/io/spine/base/Environment.java

```diff
@@ -181,50 +180,22 @@ public Environment createCopy() {
      *
      * @return the current environment type.
      */
-    @SuppressWarnings("ConstantConditions"/* no NPE is ensured by the `ensureTypeIsSet` call. */)
-    public boolean is(EnvironmentType type) {
-        ensureTypeIsSet();
-        return currentEnvType.equals(type);
-    }
-
-    /**
-     * Returns the current environment type.
-     *
-     * <p>If {@linkplain #register(EnvironmentType) custom env types have been defined},
-     * goes through them in the latest-registered to earliest-registered order.
-     * Then, checks {@link Tests} and {@link Production}.
-     *
-     * @return the current environment type
-     */
-    public EnvironmentType type() {
-        ensureTypeIsSet();
-        return currentEnvType;
-    }
-
-    private void ensureTypeIsSet() {
-        if (currentEnvType == null) {
-            determineCurrentType();
-        }
-    }
-
-    private void determineCurrentType() {
-        for (EnvironmentType type : knownEnvTypes) {
-            if (type.enabled()) {
-                this.currentEnvType = type;
-                return;
-            }
-        }
+    public boolean is(Class<? extends EnvironmentType> typeClass) {
+        EnvironmentType currentType = cachedOrCalculated();
+        boolean result = currentType.getClass()
```

Comment:
I'd go with `type.isInstance(..)`. That will give us an opportunity to create polymorphic types.